### ISSUE TITLE
slither.io is awfully slow 

### ISSUE THREAD

**Issue by ojab** [servo/servo#10381]
Minified testcase:

``` js
var fps = 0, lrd_mtm = Date.now();

function oef() {
    if (1E3 < Date.now() - lrd_mtm) {
        console.log("FPS: " + fps);
        fps = 0;
        lrd_mtm = Date.now()
    }
    redraw();
    window.requestAnimationFrame(oef)
};

function redraw() {
    fps++;
    var canvas = document.createElement("canvas");
    var canvas2 = document.createElement("canvas");
    canvas.width = 1000;
    canvas.height = 1000;
    var b = canvas2.getContext("2d");
    b.drawImage(canvas, 0, 0)
};

oef();
```

getting about 5 FPS on servo release build (with and without WebRender enabled, Mac OS X) and 60 FPS in Firefox/Safari/Chrome.
With small canvas (100x100, for example) I'm getting 60 FPS in servo.

I assume that this issue can be duplicate of #7638/#8311/#8281, but AFAIU they are about servo debug builds.

**Comment by jdm** [comment]
We should measure whether the time is being spent under `drawImage` or whether `requestAnimationFrame` isn't giving us frequent-enough callbacks.

**Comment by ojab** [comment]
It's actually drawImage() calls, 1000x1000 canvas takes about 0.160 sec w/ WebRender and 0.15-0.20 sec without.

**Comment by jdm** [comment]
I bet the solution is to switch the implementation of `draw_html_canvas_element` to use a new Canvas2dMsg which takes a handle to an existing canvas rather than the direct pixels. This will avoid the need to wait on a synchronous read-back operation from the first canvas and serialize the pixels twice.

**Comment by nox** [comment]
The solution is to not have one additional thread per canvas.

**Comment by atouchet** [comment]
This was addressed by #20680.
